docs/dev/commitlog-file-format.md

```diff
@@ -62,4 +62,55 @@ Version 2
         size            : size of all entries in this multi-entry + headers
         crc             : CRC32 of magic, size
         <entries> * N
+        crc2            : CRC32 of magic, size and data in each entry
+```
+
+Version 3
+---------
+
+Modified from v3 to improve error detection/false positive elimination.
+This version does CRC per written disk block instead of actual entry.
+Every block is also tagged with which file is being written, this to
+be able to better distinguish new data from that left over from recycling
+(reusing old files) and actual disk/file corruption.
+
+```
+
+        Segment file header
+
+        magic           : uint32_t - ('S'<<24) |('C'<< 16) | ('L' << 8) | 'C';
+        version         : uint32_t - same as descriptor
+        id              : uint64_t - same as descriptor
```

Comment:
There is no "Descriptor" defined below, so same as what?

Reply:
descriptor above. segment desc.